docs/docs/Tutorials/ROS2/CPPTemplate.md

```diff
@@ -0,0 +1,394 @@
+package_name = 'your_package_name'   # Update package name (same as in CMakeLists.txt)
+
+def generate_launch_description():
+    ld = LaunchDescription()
+    config = os.path.join(
+        get_package_share_directory(package_name),
+        'config',
+        'controller.yaml'
+        )
+
+    node = Node(
+        package=package_name,
+        name=package_name,  # ensure same as name in config.yaml
+        executable=package_name,
+```
+
+- `config/controller.yaml` (line 1)  
+    Update first line with your controller name (same as node name in launch file)
+
+``` yaml linenums="1" hl_lines="1" title="config/controller.yaml"
+/your_controller_name:
+  ros__parameters:
+    foo: 1.0
+```
+
+and rename the folder
+
+- `include/mbari_wec_template_cpp` (containing `controller.hpp` and `control_policy.hpp`) to your
+  package name
+
+resulting in the following folder structure:
+
+```
+<your_package_name>
+    ├── CMakeLists.txt
+    ├── config
+    │   └── controller.yaml
+    ├── include
+    │   └── <your_package_name>
+    │       ├── controller.hpp
+    │       └── control_policy.hpp
+    ├── launch
+    │   └── controller.launch.py
+    ├── LICENSE
+    ├── package.xml
+    ├── README.md
+    └── src
+        └── controller.cpp
+```
+
+Update the include paths in
+
+- `controller.cpp` (lines 15-16)
+``` cpp linenums="15" hl_lines="1 2" title="src/controller.cpp"
+#include <mbari_wec_template_cpp/control_policy.hpp>  // update include path
+#include <mbari_wec_template_cpp/controller.hpp>  // update include path
+```
+
+- `control_policy.hpp` (line 15)
+``` cpp linenums="15" hl_lines="1" title="include/your_package_name/control_policy.hpp"
+#include <mbari_wec_template_cpp/controller.hpp>  // update include path
+```
+
+Update include guards in
+
+- `controller.hpp`
+``` cpp linenums="15" hl_lines="1 2" title="include/your_package_name/controller.hpp"
+#ifndef YOUR_PACKAGE_NAME__CONTROLLER_HPP_
+#define YOUR_PACKAGE_NAME__CONTROLLER_HPP_
+```
+`...`
+``` cpp linenums="74" hl_lines="1"
+#endif  // YOUR_PACKAGE_NAME__CONTROLLER_HPP_
+```
+
+- `control_policy.hpp`
+``` cpp linenums="15" hl_lines="1 2" title="include/your_package_name/control_policy.hpp"
+#ifndef YOUR_PACKAGE_NAME__CONTROL_POLICY_HPP_
+#define YOUR_PACKAGE_NAME__CONTROL_POLICY_HPP_
+```
+`...`
+``` cpp linenums="67" hl_lines="1"
+#endif  // YOUR_PACKAGE_NAME__CONTROL_POLICY_HPP_
+```
+
+Modify `CMakeLists.txt` as desired and add any dependencies in `package.xml` following standard
+ROS 2 documentation.
+
+---
+
+## Implement Controller
+Assuming you have followed the above,
+
+- `include/<your_package_name>/control_policy.hpp`
+- `src/controller.cpp`
+
+are stubbed out to implement your custom external controller. You may also use
+`config/controller.yaml` for any policy parameters.
+
+### ControlPolicy
+
+You may use the struct `ControlPolicy` in `control_policy.hpp` to implement your controller.
+
+``` cpp linenums="25"
```

Comment:
Here, this box is gray, but I think it should be white, based on the pattern above. The `title` bar in the snippet boxes above helped me find the file easily.
What is the `hl_lines` in the boxes above? If it makes the appearance different, should keep the same consistency as the boxes above.
The 4 boxes below don't need a title bar, since they're just continuing the same file.

Reply:
`hl_lines` is supposed to highlight specific lines in the codeblock with line 1 being the first line of the code block, not in reference to `linenums` ... a bit confusing... The problem is, hl_lines doesn't seem to be working as advertised. I used it anyway in the hopes that it will work again in the future.

Reply:
Understood.
I think this might make the box white.
```suggestion
``` cpp linenums="25" title="include/your_package_name/control_policy.hpp"
```